### PR TITLE
Fix chain nevents

### PR DESCRIPTION
**Pull request by OlivierBondu**
aka: no need to produce 10k wmLHE for testing only nEvents GS

### CODE REVIEW

**Comment by davidsheffield** [comment]
I have been wanting to add in a way to test the matching/filter efficiency for a while. To do that, we'd need to run the wmLHE for some larger number of events. Then the GS request would need to be duplicated. One would run GEN,SIM with `nEvents` and another would be modified to just run GEN but with a larger number of events.

Also, it shouldn't take long to generate the wmLHE events (the majority of the time in the test should be spent doing SIM). Running on a larger number will give better statistics and be less likely to fail validation.

**Comment by OlivierBondu** [comment]
well sure, but the default of 10k wmLHE is 1/5th of my request... 

so I end up redoing the production itself, not really having a preliminary estimate of time and size...

**Comment by davidsheffield** [comment]
So the total number of events for your request is 2k and you don't want to run more than that? I think we can include an option to do that, although the 10k events is coming directly from McM, so presumably it will still run that many when you do validation.

**Comment by OlivierBondu** [comment]
no my request is 50k. In my understanding the McM validation jobs go to the 1nh queue, and the nEvents that goes in there to check the time and size estimates are based on what you put in the system.

In any case: I agree that we could imagine tuning the ratio wmLHE / GS according to matching/filter efficiencies and things like this, but in the end this would need development closer to the needed McM inputs, in order to not overshoot as was clearly the case for me here for wmLHE.

File: testRequests.py [hunk 1]
```diff
@@ -107,8 +107,13 @@ def getTestScript(PrepID, nEvents):
         inputfile = open(filename, 'r')
         outputfile = open(tmpfilename, 'w')
         for line in inputfile:
-            outputfile.write(re.sub('(.*RAWSIM.*-n) \d*( .*)', r'\1 %s\2'
-                                    % (nEvents), line))
+            outline = re.sub('(.*--eventcontent LHE.*-n) \d*( .*)', r'\1 %s\2'
+                                    % (nEvents), line)
+            outline = re.sub('(.*--eventcontent DQM.*-n) \d*( .*)', r'\1 %s\2'
```
Comment (davidsheffield) [inline testRequests.py:112]: Could you remind me where the cmsDriver.py commands this modifies appears. Were they in old wmLHE requests? I don't see them now. I think this is fine to include, I just want to know what command is being modified.

Comment (OlivierBondu) [inline testRequests.py:112]: I used the script to on chained request prepid, so the test job end up having 6 cmsDriver commands, I am modifying only those that are not 'harvesting' (i.e. 1,2 and 4,5 in the list below):
(If I understand the question ? please correct me if not!)

``` bash
cmsDriver.py Configuration/GenProduction/python/HIG-RunIIWinter15wmLHE-00601-fragment.py --fileout file:HIG-RunIIWinter15wmLHE-00601.root --mc --eventcontent LHE --datatier LHE --conditions MCRUN2_71_V1::All --step LHE --python_filename HIG-RunIIWinter15wmLHE-00601_1_cfg.py --no_exec --customise Configuration/DataProcessing/Utils.addMonitoring -n 100 || exit $? ;
```

``` bash
echo "nothing" ;cmsDriver.py Configuration/GenProduction/python/HIG-RunIIWinter15wmLHE-00601-fragment.py --fileout file:HIG-RunIIWinter15wmLHE-00601.root --mc --eventcontent DQM --datatier DQM --conditions MCRUN2_71_V1::All --step LHE,USER:GeneratorInterface/LHEInterface/wlhe2HepMCConverter_cff.generator,GEN,VALIDATION:genvalid_all  --fileout file:HIG-RunIIWinter15wmLHE-00601_genvalid.root --mc -n 100 --python_filename HIG-RunIIWinter15wmLHE-00601_genvalid.py --dump_python --no_exec || exit $? ;
```

``` bash
cmsDriver.py step2 --filein file:HIG-RunIIWinter15wmLHE-00601_genvalid.root --conditions MCRUN2_71_V1::All --mc -s HARVESTING:genHarvesting --harvesting AtJobEnd --python_filename HIG-RunIIWinter15wmLHE-00601_genvalid_harvesting.py --no_exec || exit $? ;
```

``` bash
cmsDriver.py Configuration/GenProduction/python/ThirteenTeV/Hadronizer_TuneCUETP8M1_13TeV_generic_LHE_pythia8_cff.py --filein file:HIG-RunIIWinter15wmLHE-00601.root --fileout file:HIG-RunIISummer15GS-00577.root --mc --eventcontent RAWSIM --customise SLHCUpgradeSimulations/Configuration/postLS1Customs.customisePostLS1,Configuration/DataProcessing/Utils.addMonitoring --datatier GEN-SIM --conditions MCRUN2_71_V1::All --beamspot Realistic50ns13TeVCollision --step GEN,SIM --magField 38T_PostLS1 --python_filename HIG-RunIISummer15GS-00577_1_cfg.py --no_exec -n 100 || exit $? ;
```

``` bash
echo "nothing" ;cmsDriver.py Configuration/GenProduction/python/ThirteenTeV/Hadronizer_TuneCUETP8M1_13TeV_generic_LHE_pythia8_cff.py --filein file:HIG-RunIIWinter15wmLHE-00601.root --fileout file:HIG-RunIISummer15GS-00577.root --mc --eventcontent DQM --datatier DQM --conditions MCRUN2_71_V1::All --beamspot Realistic50ns13TeVCollision --step GEN,VALIDATION:genvalid_all --magField 38T_PostLS1  --fileout file:HIG-RunIISummer15GS-00577_genvalid.root --mc -n 100 --python_filename HIG-RunIISummer15GS-00577_genvalid.py  --no_exec || exit $? ;
```

``` bash
cmsDriver.py step2 --filein file:HIG-RunIISummer15GS-00577_genvalid.root --conditions MCRUN2_71_V1::All --mc -s HARVESTING:genHarvesting --harvesting AtJobEnd --python_filename HIG-RunIISummer15GS-00577_genvalid_harvesting.py --no_exec || exit $? ;
```

Comment (davidsheffield) [inline testRequests.py:112]: OK. I was just looking at a chain with fewer cmsDriver.py commands in it.

**Comment by davidsheffield** [comment]
I think this is fine for now. As I said, I'd like to change it to automatically get efficiencies at some point. When I do, I'd probably make this behavior of only running on `nEvents` for everything an option that you could access with a flag.

**Comment by davidsheffield** [comment]
Thanks for adding to these scripts, @OlivierBondu.

**Comment by OlivierBondu** [comment]
welcome :)

**Pull request merged**